vm/src/memory/operand_stack.cpp

```diff
@@ -54,14 +55,26 @@ const char *OperandStack::pop_bytes(size_t n) {
     if (current_pos < n) {
         throw OperandStackUnderflowError("operand stack is empty");
     }
+    operands_refs.resize(operands_refs.size() - n);
     current_pos -= n;
     return bytes + current_pos;
 }
 
+void OperandStack::transfer(OperandStack &caller_stack, size_t n) {
+#ifndef NDEBUG
+    if (n > caller_stack.size())
+        throw std::out_of_range("cannot transfer more byte than contained in source operand stack");
+#endif
+    memcpy(this->bytes + current_pos, caller_stack.bytes + (caller_stack.size() - n), n);
```

Comment:
Given that this function could accept the same instance, it would be great to document to not to. `memcpy` doesn't play well if the two pointers point to overlapping memory regions.